### PR TITLE
Add support ipv6 configuration in azurerm_express_route_circuit_peering

### PR DESCRIPTION
**Pull request by xuzhang3**
AccTests Result:
image

### CODE REVIEW

File: azurerm/internal/services/network/express_route_circuit_peering_resource.go [hunk 1]
```diff
@@ -108,6 +108,54 @@ func resourceArmExpressRouteCircuitPeering() *schema.Resource {
...
+						"microsoft_peering_config": {
+							Type:     schema.TypeList,
+							Required: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"advertised_public_prefixes": {
+										Type:     schema.TypeList,
+										Optional: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
```
Comment (katbyte) [inline azurerm/internal/services/network/express_route_circuit_peering_resource.go:126]: could we validate this?

File: azurerm/internal/services/network/tests/express_route_circuit_peering_resource_test.go [hunk 2]
```diff
@@ -311,6 +368,188 @@ resource "azurerm_express_route_circuit_peering" "test" {
...
+}
+
+func testAccAzureRMExpressRouteCircuitPeering_msPeeringIpv6WithRouteFilter(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
```
Comment (katbyte) [inline azurerm/internal/services/network/tests/express_route_circuit_peering_resource_test.go:489]: could we use this for the resource group names:
```suggestion
  name     = "acctestRG-expressroute-%d"
```

File: website/docs/r/express_route_circuit_peering.html.markdown [hunk 3]
```diff
@@ -82,6 +92,14 @@ A `microsoft_peering_config` block contains:
 * `customer_asn` - (Optional) The CustomerASN of the peering
 * `routing_registry_name` - (Optional) The RoutingRegistryName of the configuration
 
+A `ipv6_peering` block contains:
+
+* `microsoft_peering_config` - (Required)  A `microsoft_peering_config` block as defined above.
```
Comment (katbyte) [inline website/docs/r/express_route_circuit_peering.html.markdown:97]: I don't see this block defined anywhere?

File: azurerm/internal/services/network/express_route_circuit_peering_resource.go [hunk 4]
```diff
@@ -108,6 +108,54 @@ func resourceArmExpressRouteCircuitPeering() *schema.Resource {
 				},
 			},
 
+			"ipv6_peering": {
```
Comment (katbyte) [inline azurerm/internal/services/network/express_route_circuit_peering_resource.go:111]: given the resource name i think peering is implied?
```suggestion
			"ipv6": {
```

File: azurerm/internal/services/network/express_route_circuit_peering_resource.go [hunk 5]
```diff
@@ -108,6 +108,54 @@ func resourceArmExpressRouteCircuitPeering() *schema.Resource {
...
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"microsoft_peering_config": {
```
Comment (katbyte) [inline azurerm/internal/services/network/express_route_circuit_peering_resource.go:117]: can we just shorten this to
```suggestion
						"microsoft_peering": {
```

File: azurerm/internal/services/network/express_route_circuit_peering_resource.go [hunk 6]
```diff
@@ -320,6 +377,26 @@ func expandExpressRouteCircuitPeeringMicrosoftConfig(input []interface{}) *netwo
...
+		return nil
+	}
+
+	v := input[0].(map[string]interface{})
+	peeringConfig := network.Ipv6ExpressRouteCircuitPeeringConfig{
+		PrimaryPeerAddressPrefix:   utils.String(v["primary_peer_address_prefix"].(string)),
+		SecondaryPeerAddressPrefix: utils.String(v["secondary_peer_address_prefix"].(string)),
+		MicrosoftPeeringConfig:     expandExpressRouteCircuitPeeringMicrosoftConfig(v["microsoft_peering_config"].([]interface{})),
+	}
+	routeFilterId := v["route_filter_id"].(string)
```
Comment (katbyte) [inline azurerm/internal/services/network/express_route_circuit_peering_resource.go:391]: could we add validation to check this is not set when incompatible types are used?

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @xuzhang3 - overall this looks good and i've left some comments inline to address

File: website/docs/r/express_route_circuit_peering.html.markdown (hunk 3 above)
Comment (xuzhang3) [reply website/docs/r/express_route_circuit_peering.html.markdown]: Rename `microsoft_peering_config` to `microsoft_peering` and  add a new `microsoft_peering` block description

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @xuzhang3 - left some more comments inline

File: azurerm/internal/services/network/express_route_circuit_peering_resource.go (hunk 6 above)
Comment (xuzhang3) [reply azurerm/internal/services/network/express_route_circuit_peering_resource.go:399]: I don't know why GitHub does not display the latest code.  I have add a check for `routeFilterId `. https://github.com/xuzhang3/terraform-provider-azurerm/blob/f/expressroutecircuitpeering_enhance/azurerm/internal/services/network/express_route_circuit_peering_resource.go#L400

File: website/docs/r/express_route_circuit_peering.html.markdown [hunk 7]
```diff
@@ -63,24 +72,57 @@ The following arguments are supported:
...
+
+---
+
+A `microsoft_peering` block contains:
+
+* `advertised_public_prefixes` - (Required) A list of Advertised Public Prefixes.
+
+* `customer_asn` - (Optional) The CustomerASN of the peering.
+
+* `routing_registry_name` - (Optional) The Routing Registry against which the AS number and prefixes are registered. For example: ARIN, RIPE, AFRINIC etc.
```
Comment (katbyte) [inline website/docs/r/express_route_circuit_peering.html.markdown:123]:
```suggestion
* `routing_registry_name` - (Optional) The Routing Registry against which the AS number and prefixes are registered. For example: `ARIN`, `RIPE`, `AFRINIC` etc. 
```

File: website/docs/r/express_route_circuit_peering.html.markdown [hunk 8]
```diff
@@ -63,24 +72,57 @@ The following arguments are supported:
...
 A `microsoft_peering_config` block contains:
 
-* `advertised_public_prefixes` - (Required) A list of Advertised Public Prefixes
-* `customer_asn` - (Optional) The CustomerASN of the peering
-* `routing_registry_name` - (Optional) The RoutingRegistryName of the configuration
+* `advertised_public_prefixes` - (Required) A list of Advertised Public Prefixes.
+
+* `customer_asn` - (Optional) The CustomerASN of the peering.
+
+* `routing_registry_name` - (Optional) The Routing Registry against which the AS number and prefixes are registered.  For example: ARIN, RIPE, AFRINIC etc.
```
Comment (katbyte) [inline website/docs/r/express_route_circuit_peering.html.markdown:101]:
```suggestion
* `routing_registry_name` - (Optional) The Routing Registry against which the AS number and prefixes are registered.  For example: `ARIN`, `RIPE`, `AFRINIC` etc. 
```

**Review by katbyte — CHANGES_REQUESTED** [review]
Hey @xuzhang3 - i still don't see a check to ensure route filter id is not set when the type is incorrect

File: azurerm/internal/services/network/express_route_circuit_peering_resource.go (hunk 6 above)
Comment (xuzhang3) [reply azurerm/internal/services/network/express_route_circuit_peering_resource.go:413]: `ipv6`  is a little different with `microsoft_peering_config`, it is only supported by `MicrosoftPeering`.  I add a check for `ipv6`, if peering type is not `MicrosoftPeering` and `ipv6` is configured, an error message will raise up.
Check `ipv6` configure:
https://github.com/xuzhang3/terraform-provider-azurerm/blob/f/expressroutecircuitpeering_enhance/azurerm/internal/services/network/express_route_circuit_peering_resource.go#L267

**Review by katbyte — APPROVED** [review]
Thanks @xuzhang3 - LGTM 👍
